bsyncpy/bsyncpy_generator.py

```diff
@@ -374,16 +373,45 @@ def do_complexType(element) -> BSElement:
     elif choice is not None:
         logging.debug("    choice!")
         for i, child in enumerate(choice):
-            child_name = child.get("name")
-            child_ref = child.get("ref")
-            logging.debug(f"    c [{i}] {child} {child_name} {child_ref}")
+            logging.debug(f"    c [{i}] {child} {child.get('name')}")
 
-            choice_element = do_element(child)
-            logging.debug(f"        - choice_element: {choice_element.element_name}")
+            if child.tag == "element":
+                child_name = child.get("name")
+                child_ref = child.get("ref")
 
-            bs_element.element_children.append(
-                (child_name, choice_element.element_full_name)
-            )
+                if child_name and child_ref:
```

Comment:
This was the chunk of code that was needed to resolve the reference not found issue. In short, we were not checking for references in choice elements.